### PR TITLE
Add dispute readiness overview card

### PR DESCRIPTION
**Pull request by dmallory42**
Fixes RSM-1967

#### Changes proposed in this Pull Request

Adds a feature-flagged Dispute readiness card to Payments Overview.

This PR includes:

* A new `_wcpay_feature_dispute_readiness_overview` feature flag, exposed to the client as `wcpaySettings.featureFlags.isDisputeReadinessOverviewEnabled`.
* A shared server-side dispute readiness service that evaluates refund policy, terms, statement descriptor, and support contact signals.
* New REST endpoints:
  * `GET /wc/v3/payments/dispute-readiness`
  * `POST /wc/v3/payments/dispute-readiness/dismiss`
* Store-wide dismissal/reappearance metadata.
* A new client data slice and Payments Overview card.
* Tracks events for card views, CTA clicks, and dismissals.
* PHP and JS coverage for the service, REST controller, resolver, and dismiss action.

Screenshot 2026-05-26 at 13 49 35

#### Testing instructions

This test plan is for testing the PR build on a brand-new store. No command-line steps are required.

##### 1. Create a new test store

1. Open the Jurassic Ninja launch link from the build comment: https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/dispute-readiness-scores
2. Wait for the site to finish creating.
3. Log in to the WordPress dashboard when prompted.

##### 2. Set up WooPayments enough to reach the overview

1. In the WordPress dashboard, go to **Payments → Overview**.
2. Complete any required WooPayments setup steps until the Payments overview page loads.
3. If you see a test/development mode notice, that is expected for this test site.

##### 3. Check that the Dispute Readiness card appears

1. On **Payments → Overview**, look below the **Balance** card and the **Account details** card.
2. Confirm there is a card titled **Dispute Readiness**.
3. Confirm the card has the same general styling as the other overview cards, including a header area with a thin divider underneath.
4. Confirm the card text says something like: “These 4 steps help customers recognize charges, understand your policies, and contact you before opening a dispute.”
5. Confirm the card shows a circular progress indicator, for example **2 of 4**, **3 of 4**, or **4 of 4**.

##### 4. Check the checklist items

1. Confirm the card lists these four items:
   - Recognizable statement descriptor
   - Refund policy page published
   - Customer support contact linked in order emails
   - Terms & conditions linked at checkout
2. Confirm completed items show a green check mark.
3. Confirm incomplete items show a red X.
4. Confirm incomplete items with a next step have a **Fix →** or **Review →** link aligned on the right.

##### 5. Test one fix flow

1. Find an incomplete checklist item with a **Fix →** or **Review →** link.
2. Click the link.
3. Confirm it takes you to a relevant settings or page-editing screen.
4. Make a simple change if possible. For example:
   - Add or publish a refund policy page, or
   - Add or publish a terms and conditions page.
5. Return to **Payments → Overview**.
6. Confirm the Dispute Readiness card updates after the page reloads.

##### 6. Test the Learn more link

1. Click **Learn more →** in the Dispute Readiness card.
2. Confirm it opens WooPayments dispute documentation in a new browser tab.
3. Close the new tab and return to the test site.

##### 7. Test dismissing the card

1. Click the **X** in the top-right corner of the Dispute Readiness card.
2. Confirm the card disappears.
3. Refresh the **Payments → Overview** page.
4. Confirm the card stays hidden after refresh.

##### Expected result

The Dispute Readiness card should appear on a new store’s Payments overview page, match the styling of the other overview cards, clearly show the store’s dispute readiness progress, link to useful next steps, and stay dismissed after the user closes it.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times.
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: src/Internal/Service/DisputeReadinessService.php [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+	 * Returns cached account data safely.
+	 *
+	 * @return array
+	 */
+	private function get_cached_account_data(): array {
+		if ( ! class_exists( '\WC_Payments' ) || ! \WC_Payments::get_account_service() ) {
+			return [];
+		}
+
+		$account_data = \WC_Payments::get_account_service()->get_cached_account_data();
```
Comment (mgascam) [inline src/Internal/Service/DisputeReadinessService.php]: Was there a reason to reach for `WC_Payments::get_account_service()` statically here instead of injecting `WC_Payments_Account` via the constructor? `Level3Service` next door does the latter, and `LegacyContainer.php` already exposes `WC_Payments_Account` to the container for this case. Injecting it would also make the service easier to mock in tests.

Something like:
`public function __construct( \WC_Payments_Account $account ) { ... }` with `->addArgument( WC_Payments_Account::class )` on the registration in GenericServiceProvider.

File: client/overview/dispute-readiness/style.scss [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	padding: 0;
+	color: #50575e;
+	font-size: 24px;
+	line-height: 24px;
+	text-decoration: none;
+
+	&:hover,
+	&:focus {
+		color: #1e1e1e;
+		box-shadow: none;
```
Comment (mgascam) [inline client/overview/dispute-readiness/style.scss]: Claude flagged a potential a11y issue. 

The &:hover, &:focus rule (lines 27-31) sets `box-shadow: none`, zeroing the WP Button focus indicator. The only remaining cue is a color shift on the × glyph, and nothing renders in forced-colors mode.

Either drop the box-shadow: none override and let the WP Button ring render, or replace it explicitly:

```
  &:focus,
  &:focus-visible {
      outline: 2px solid var(--wp-admin-theme-color, #3858e9);
      outline-offset: 2px;
      box-shadow: none;
  }
```

File: client/overview/dispute-readiness/index.tsx
Comment (mgascam) [inline client/overview/dispute-readiness/index.tsx:1]: Should we add a unit test for this component?  I'd suggest the following tests: it renders null when disabled, renders with correct score, dismiss click invokes the action, renders null when isDismissed.

File: client/data/dispute-readiness/reducer.ts
Comment (mgascam) [inline client/data/dispute-readiness/reducer.ts:1]: I noticed  `reducer.ts`, s`electors.ts`, and `hooks.ts` all ship without tests, not sure if it is intended

**Review by mgascam — COMMENTED** [review]
Hey @dmallory42, nice work on this! Left a few comments. The only one I'd ask you to look at before merging is the one about injecting the `WC_Payments_Account`. The rest are suggestions worth double-checking.

File: src/Internal/Service/DisputeReadinessService.php (hunk 1 above)
Comment (dmallory42) [reply src/Internal/Service/DisputeReadinessService.php]: Good call — updated this to inject `WC_Payments_Account` through the constructor and register the service with `WC_Payments_Account::class` in `GenericServiceProvider`. The REST/controller usage now resolves `DisputeReadinessService` from the container, and the tests mock the injected account service directly.

File: client/overview/dispute-readiness/style.scss (hunk 2 above)
Comment (dmallory42) [reply client/overview/dispute-readiness/style.scss]: Addressed. I added an explicit accessible focus style for the dismiss button so it no longer relies only on the glyph color change and still has a visible focus indicator.

File: client/overview/dispute-readiness/index.tsx
Comment (dmallory42) [reply client/overview/dispute-readiness/index.tsx:1]: Added component coverage for the Dispute Readiness card, including disabled/null rendering, score rendering, dismiss action behavior, and dismissed/null rendering.

File: client/data/dispute-readiness/reducer.ts
Comment (dmallory42) [reply client/data/dispute-readiness/reducer.ts:1]: Added tests for the new dispute readiness data layer, including reducer and selector coverage. The existing action/resolver coverage was also kept with the TypeScript conversion.

**Comment by dmallory42** [comment]
Thanks for the review, @mgascam! This is ready for another pass, when you get a chance.

**Review by mgascam — APPROVED** [review]
Thanks for addressing my comments @dmallory42 :shipit:

**Comment by lucyneb** [comment]
@dmallory42 looks good!
